lib/appsignal/hooks/at_exit.rb

```diff
@@ -31,7 +38,8 @@ def self.call
           Appsignal.report_error(error) do |transaction|
             transaction.set_namespace("unhandled")
           end
-          Appsignal.stop("at_exit")
+        ensure
+          Appsignal.stop("at_exit") if Appsignal.config&.[](:enable_at_exit_hook)
```

Comment:
I think I'll add back the behavior to always call `Appsignal.stop` when it encounters an error, just in case I don't break any existing behavior apps rely on.

So something like:

```suggestion
          Appsignal.stop("at_exit") if Appsignal.config&.[](:enable_at_exit_hook) || error
```